leader/backoff.go

```diff
@@ -1,52 +1,100 @@
 package leader
 
 import (
-	"math"
-	"math/rand"
 	"time"
 
-	"github.com/cenk/backoff"
+	"github.com/cenkalti/backoff"
 )
 
-// FreebieExponentialBackOff
-type FreebieExponentialBackOff struct {
-	InitialInterval     time.Duration
-	RandomizationFactor float64
-	Multiplier          float64
-	MaxInterval         time.Duration
-	MaxElapsedTime      time.Duration
+// NewCountingBackOff returns a new instance of the CountingBackOff
+// with the specified backoff as implementation
+func NewCountingBackOff(backOff backoff.BackOff) *CountingBackOff {
```

Comment:
this is all too complex, can we replace all this thing with something else, even static 1 second delay + randomizer will work here.